.github/workflows/bench.yml

```diff
@@ -52,7 +52,7 @@ jobs:
             ftype: q4_0
             pr_comment_enabled: "true"
 
-    if: ${{ github.event.inputs.gpu-series == 'Standard_NC4as_T4_v3' || github.event.schedule || github.event.pull_request || github.head_ref == 'master' || github.ref_name == 'master' || github.event.push.ref == 'refs/heads/master' }}
+    if: ${{ github.event.inputs.gpu-series == 'Standard_NC4as_T4_v3' || (github.repository_owner == 'ggerganov' && github.event.schedule) || github.event.pull_request || github.head_ref == 'master' || (github.event.push && (github.repository_owner == 'ggerganov' && github.event.push.ref == 'refs/heads/master') || github.ref_name == 'master') }}
```

Comment:
Maybe you can try writing the condition in multiple lines, that should be easier to spot errors. Here's how to do: https://github.com/orgs/community/discussions/25641

The current condition when breaking into multiple line:

```
github.event.inputs.gpu-series == 'Standard_NC4as_T4_v3'
|| (github.repository_owner == 'ggerganov' && github.event.schedule)
|| github.event.pull_request
|| github.head_ref == 'master'
|| (
  github.event.push && (
    github.repository_owner == 'ggerganov' && github.event.push.ref == 'refs/heads/master'
  )
  || github.ref_name == 'master'
)
```

Which according to [Boolean algebra](https://en.wikipedia.org/wiki/Boolean_algebra) means that `github.ref_name == 'master'` can be moved out of the parenthese, so I'm not sure if it's expected or not.

In anyway, here is another version which should be equivalent (using boolean algebra - **keep in mind that I'm not 100% sure, please verify again**):

```
github.event.inputs.gpu-series == 'Standard_NC4as_T4_v3'
|| github.event.pull_request
|| github.head_ref == 'master'
|| github.ref_name == 'master'
|| (
  github.repository_owner == 'ggerganov'
  && (
    github.event.schedule
    || (github.event.push && github.event.push.ref == 'refs/heads/master')
  )
)
```

Reply:
The `github.ref_name` check is part of a [ternary](https://docs.github.com/en/actions/learn-github-actions/expressions#example) as it should be excluded when github.event.push is truthy.

Reply:
Upon reading the docs again I'm not sure if that quite works though. :(

Since the docs say it will actually evaluate the first statement after `&&` and fall back to the statement after `||` if falsy (ie, not quite a real ternary) it might be better to ditch the ternary and check github.event.push twice instead.

Reply:
Ternary operator does not work in this case. Take this equation for example: `a && b || c`. Using ternary, we expect it to behave like `a ? b : c`.

Now in the case where `a = true`, `b = false`, `c = true`:
- `a ? b : c` returns `false`
- `a && b || c` returns `true`

Reply:
Yeah, pretty much what I just discovered myself. :P